roles/testnode/README.rst

```diff
@@ -0,0 +1,215 @@
+
+A list of packages to upgrade. These lists are defined in the vars files in ``vars/``::
+
+    packages_to_upgrade: []
+
+The user that teuthology will use to connect to testnodes. This user will be created by this role and assigned to the appropriate groups.
+Eventhough this variable exists, teuthology is not quite ready to support a configurable user::
+
+    teuthology_user: "ubuntu"
+
+This user is created for use in running xfstests from ceph-qa-suite::
+
+    xfstests_user: "fsgqa"
+
+This will control whether or not rpcbind is started before nfs.  Some distros require this, others don't::
+
+    start_rpcbind: true
+
+Set to true if /etc/fstab must be modified to persist things like mount options, which is useful for long lived
+bare metal machines, less useful for virtual machines that are re-imaged before each job::
+
+    modify_fstab: true
+
+A list of ntp servers to use::
+
+    ntp_servers:
+      - 0.us.pool.ntp.org
+      - 1.us.pool.ntp.org
+      - 2.us.pool.ntp.org
+      - 3.us.pool.ntp.org
+
+The lab domain to strip from hostname on rhel 7.x nodes. This is only done if ``lab_domain``
+is defined::
+
+    lab_domain: ''
+
+Tags
+++++
+
+Available tags are listed below:
+
+cpan
+    Install and configure cpan and Amazon::S3.
+
+entitlements
+    Configure Red Hat entitlements with subscription-manager. Inherited from the common role.
+
+epel
+    Create and manage /etc/yum.repos.d/epel.repo. Inherited from the common role.
+
+gpg-keys
+    Install gpg keys on Fedora.    
+
+hostname
+    Strip ``lab_domain`` from the hostname on Rhel 7.
+
+kerberos
+    Install and Configure a Kerberos client. Inherited from the common role.
+
+kernel_logging
+    Runs a script that enabled kernel logging to the console on ubuntu.        
+
+monitoring-scripts
+    Uploads custom disk monitoring scripts. See, ``tasks/disk_monitoring.yml``.
+
+nagios
+    Configure nagios nrpe server for apt systems. 
+
+nfs
+    Install and start nfs.
+
+ntp-client
+    Setup ntp.
+
+packages
+    Install, update and remove packages.
+
+pip
+    Install and configure pip.
+
+pubkeys
+    Adds the ssh public keys for the ``teuthology_user``.    
+
+remove-ceph
+    Ensure all ceph related packages are removed. See ``packages_to_remove`` in the distros var file for the list.    
+
+repos
+    Perform all repo related tasks. Creates and manages our custom repo files.     
+
+selinux
+    Configure selinux on yum systems.    
+
+ssh
+    Manage things ssh related.  Will upload the distro specific sshd_config, ssh_config and addition of pubkeys for the ``teuthology_user``. 
+
+sudoers
+    Manage the /etc/sudoers and the nagios suders.d files.
+
+user
+    Manages the ``teuthology_user`` and ``xfstests_user``. 
+
+Dependencies
+++++++++++++
+
+This role depends on the following roles:
+
+secrets
+    Provides a var, ``secrets_path``, containing the path to where our secrets are located.
+    
+sudo
+    Sets ``ansible_sudo: true`` for this role.
```

Comment:
...which causes all the plays "in" (is that the right term) this role to execute with sudo